event/event.go

```diff
@@ -239,11 +241,27 @@ type ReleaseEventCommon struct {
 	Error string `json:"error,omitempty"`
 }
 
+const (
+	// ReleaseImageSpecType is a type of release spec when there are update.Images
+	ReleaseImageSpecType = "releaseImageSpecType"
+	// ReleaseContainersSpecType is a type of release spec when there are update.Containers
+	ReleaseContainersSpecType = "releaseContainersSpecType"
+)
+
+// ReleaseSpec is a spec for images and containers release
+type ReleaseSpec struct {
+	// Type is ReleaseImageSpecType or ReleaseContainersSpecType
+	// if empty (for previous version), then use ReleaseImageSpecType
+	Type                  string
+	ReleaseImageSpec      *update.ReleaseImageSpec
+	ReleaseContainersSpec *update.ReleaseContainersSpec
+}
+
 // ReleaseEventMetadata is the metadata for when service(s) are released
 type ReleaseEventMetadata struct {
 	ReleaseEventCommon
-	Spec  update.ReleaseSpec `json:"spec"`
-	Cause update.Cause       `json:"cause"`
+	Spec  ReleaseSpec  `json:"spec"`
```

Comment:
I am not convinced that this will parse old serialisations of this struct. My reasoning goes like this:

This field used to contain what is now `update.ReleaseImageSpec`, so would get serialised as

```
{
  "revision": ..., "result", ..., "error", ..., // from ReleaseEventCommon
  "spec": {
    "serviceSpecs": ...,
    "imageSpec": ...,
    // and so on, from update.ReleaseSpec
  },
```

Now, the equivalent value would serialise as

```
{
  "revision", // etc. from ReleaseEventCommon
  "spec": {
    "type": "releaseImageSpecType",
    "releaseImageSpec": {
      "serviceSpecs": ..., // etc.
    },
```

and, a value of the first kind would now be decoded as

```
ReleaseEventMetadata{
  ReleaseEventCommon: ReleaseEventCommon{Revision: ...etc},
  Spec: ReleaseSpec{Type: "", ReleaseImageSpec: nil, ReleaseContainerSpec: nil},
}
```

I _think_ the tweak needed is to make `ReleaseImageSpec` and `ReleaseContainerSpec` both embedded, i.e,

```
type ReleaseSpec struct {
   Type string
    *update.ReleaseImageSpec
    *update.ReleaseContainerSpec
}
```

I may have this wrong of course -- the JSON codec often trips me up. The way to check (or demonstrate that I'm wrong!) would be to take a value serialised from the code in master, and decode it with the code here -- as a test, even.